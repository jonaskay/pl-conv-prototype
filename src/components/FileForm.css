.FileForm {
  text-align: center;
}

.FileForm-form {
  display: flex;
  width: 100%;
}

.FileForm-input {
  background: #616161;
  border: none;
  border-radius: .25rem;
  box-shadow: 
    0 1px 3px 0 rgba(0,0,0,.2), 
    0 1px 1px 0 rgba(0,0,0,.14), 
    0 2px 1px -1px rgba(0,0,0,.12);
  color: rgba(255,255,255,0.7);
  flex: auto;
  margin-right: .5rem;
  padding: .5rem;
}

.FileForm-button {
  background: #536DFE;
  border: none;
  border-radius: .25rem;
  box-shadow: 
    0 1px 3px 0 rgba(0,0,0,.2), 
    0 1px 1px 0 rgba(0,0,0,.14), 
    0 2px 1px -1px rgba(0,0,0,.12);
  color: #FFF;
  cursor: pointer;
  flex: none;
  font-weight: 700;
  outline: none;
  padding: .5rem .75rem;
}
