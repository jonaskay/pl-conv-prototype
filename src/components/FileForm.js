import React from 'react';
import './FileForm.css';

class FileForm extends React.Component {
  render() {
    return (
      <div className="FileForm">
        <div style={{fontSize: '1.25rem', marginBottom: '1rem'}}>New Playlist</div>
        <div style={{marginBottom: '1rem'}}>Choose a playlist file to start</div>
        <form className="FileForm-form" onSubmit={this.props.onSubmit}>
          <input
            type="text"
            value={this.props.fileName}
            className="FileForm-input"
            disabled />
          <input type="submit" value=">" className="FileForm-button" />
        </form>
      </div>
    );
  }
}

export default FileForm;
