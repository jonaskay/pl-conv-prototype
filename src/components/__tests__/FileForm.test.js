import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import FileForm from '../FileForm';

Enzyme.configure({ adapter: new Adapter() });

it('renders a form', () => {
  const component = shallow(<FileForm />);
  expect(component).toMatchSnapshot();
});

it('handles form submit', () => {
  const mockFunc = jest.fn();
  const component = shallow(<FileForm onSubmit={mockFunc} />);
  component.find('.FileForm-form').simulate('submit');

  expect(mockFunc).toBeCalled();
})
