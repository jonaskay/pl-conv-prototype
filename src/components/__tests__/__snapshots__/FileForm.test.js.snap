// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a form 1`] = `
<div
  className="FileForm"
>
  <div
    style={
      Object {
        "fontSize": "1.25rem",
        "marginBottom": "1rem",
      }
    }
  >
    New Playlist
  </div>
  <div
    style={
      Object {
        "marginBottom": "1rem",
      }
    }
  >
    Choose a playlist file to start
  </div>
  <form
    className="FileForm-form"
  >
    <input
      className="FileForm-input"
      disabled={true}
      type="text"
    />
    <input
      className="FileForm-button"
      type="submit"
      value=">"
    />
  </form>
</div>
`;
